R/pkg/R/utils.R

```diff
@@ -689,3 +689,7 @@ getSparkContext <- function() {
   sc <- get(".sparkRjsc", envir = .sparkREnv)
   sc
 }
+
+is_master_local <- function(master) {
+  grepl("^local(\\[[0-9\\*]+\\])?$", master, perl = TRUE)
```

Comment:
Ah sorry, I missed that... I guess it shouldn't match `local[]` based on [master url](https://spark.apache.org/docs/latest/submitting-applications.html#master-urls) so perhaps `^local(\\[([0-9]+|\\*+)\\])?$`